openapi/schemas/processes-dru/cwl.yaml

```diff
@@ -0,0 +1,1724 @@
+      - $comment: Additional properties added here to avoid error by any of the other object individually.
+        type: object
+        additionalProperties: false
+  CWLHintsList:
+    type: array
+    title: CWLHintsList
+    items:
+      $ref: '#/$definitions/CWLHintsItem'
+  CWLHintsItem:
+    title: CWLHintsItem
+    oneOf:
+    - $ref: '#/$definitions/cwltool:CUDARequirementItem'
+    - $ref: '#/$definitions/DockerRequirementItem'
+    - $ref: '#/$definitions/InitialWorkDirRequirementItem'
+    - $ref: '#/$definitions/InlineJavascriptRequirementItem'
+    - $ref: '#/$definitions/InplaceUpdateRequirementItem'
+    - $ref: '#/$definitions/LoadListingRequirementItem'
+    - $ref: '#/$definitions/NetworkAccessRequirementItem'
+    - $ref: '#/$definitions/ResourceRequirementItem'
+    - $ref: '#/$definitions/ScatterFeatureRequirementItem'
+    - $ref: '#/$definitions/ToolTimeLimitRequirementItem'
+    - $ref: '#/$definitions/WorkReuseRequirementItem'
+    - $ref: '#/$definitions/BuiltinRequirementItem'
+    - $ref: '#/$definitions/OGCAPIRequirementItem'
+    - $ref: '#/$definitions/WPS1RequirementItem'
+    - $ref: '#/$definitions/UnknownRequirementItem'
+  CWLCommand:
+    oneOf:
+    - type: string
+      title: String command.
+    - $ref: '#/$definitions/CommandParts'
+    title: CWLCommand
+    description: |
+      Command called in the docker image or on shell according to requirements
+      and hints specifications. Can be omitted if already defined in the docker
+      image.
+  CommandParts:
+    type: array
+    title: Command Parts
+    items:
+      type: string
+      title: cmd
+  CWLArgumentObject:
+    type: object
+    properties:
+      prefix:
+        type: string
+      position:
+        type: integer
+      valueFrom:
+        type: string
+    additionalProperties: {}
+  CWLArguments:
+    type: array
+    title: CWLArguments
+    description: Base arguments passed to the command.
+    items:
+      oneOf:
+      - type: string
+      - $ref: '#/$definitions/CWLArgumentObject'
+  CWLInputStdInDefinition:
+    description: |
+      Indicates that the value passed to this CWL input will be redirected to the standard input stream of the command.
+      Can be defined for only one input and must not be combined with 'stdin' at the root of the CWL document.
+    type: string
+    enum:
+      - stdin
+  CWLInputStdInObjectType:
+    type: object
+    properties:
+      type:
+        $ref: '#/$definitions/CWLInputStdInDefinition'
+    required:
+      - type
+  CWLInputStdIn:
+    oneOf:
+      - $ref: '#/$definitions/CWLInputStdInDefinition'
+      - $ref: '#/$definitions/CWLInputStdInObjectType'
+  CWLOutputStdOutDefinition:
+    description: |
+      Indicates that the data pushed to the standard output stream by the command will be redirected to this CWL output.
+      Can be defined for only one output and must not be combined with 'stdout' at the root of the CWL document.
```

Comment:
Ok. Just to confirm I understood it.
The top-level `stdout` will set the name/location of the file, and the type `stdout` will indicate which output writes to it?

Reply:
That's basically correct, yes. The use of a top-level `stdout` does not require the use of a `type: stdout` in the outputs section. That stdout file could be consumed by a matching `outputBinding: { glob: … }` construct of a `type: File` or `type: Directory` or similar; or it may be unconsumed entirely.